### PR TITLE
External authorization promotion

### PR DESCRIPTION
**Pull request by bochengchu**
Promote external authorization to beta.

Performance/benchmark tests: https://github.com/istio/tools/tree/master/perf/benchmark/configs/istio/ext-authz

### CODE REVIEW

File: features/external_authorization.md
Comment (bochengchu) [reply features/external_authorization.md:157]: https://docs.google.com/document/d/1V4mCQCw7mlGp0zSQQXYoBdbKMDnkPOjeyUb85U07iSI/edit?usp=sharing

File: features.yaml
Comment (ericvn) [inline features.yaml:242]: Update to point to checklist
```suggestion
      checklist: features/external_authorization.md
```

File: features/external_authorization.md
Comment (ericvn) [inline features/external_authorization.md:159]:
```suggestion
	relevant work group leads.  [Design Doc](https://docs.google.com/document/d/1V4mCQCw7mlGp0zSQQXYoBdbKMDnkPOjeyUb85U07iSI/edit).
```

File: features/external_authorization.md
Comment (ericvn) [inline features/external_authorization.md:160]:
```suggestion
- [x] Feature coverage and test plans written and approved.  [Test Plan](https://docs.google.com/document/d/1V4mCQCw7mlGp0zSQQXYoBdbKMDnkPOjeyUb85U07iSI/edit#heading=h.mj0rzx8l17zl).
```

File: features/external_authorization.md
Comment (ericvn) [inline features/external_authorization.md:165]: Correct links?
```suggestion
- [x] Documentation on istio.io includes samples/tutorials. [Here](https://istio.io/latest/docs/tasks/security/authorization/authz-custom/)
and [Here](https://istio.io/latest/blog/2021/better-external-authz/).
```

File: features/external_authorization.md
Comment (ericvn) [inline features/external_authorization.md:168]:
```suggestion
- [x] Release notes have been added. [Here](https://istio.io/latest/news/releases/1.9.x/announcing-1.9/#integration-with-external-authorization-systems-experimental) and [Here](https://istio.io/latest/news/releases/1.9.x/announcing-1.9/change-notes/).
```

File: features/external_authorization.md
Comment (ericvn) [inline features/external_authorization.md:199]: Since the TOC is reviewing this PR, I believe this can be checked as the will merge when there are sufficient TOC approvals.
```suggestion
- [x] The TOC has reviewed and approved promotion of the feature as part of the
```

**Review by ericvn — COMMENTED** [review]
LGTM - Approved. Several nits. I'd update the doc, pointing to some of the links.

File: features/external_authorization.md
Comment (ericvn) [reply features/external_authorization.md:165]: While updating the header, I would maybe tweak/sync the names in the header and features.yaml and add a short description.

Comment (bochengchu) [reply features/external_authorization.md:165]: Added link to the task.

File: features/external_authorization.md
Comment (therealmitchconnors) [inline features/external_authorization.md:165]: I don't see any performance expectations listed in https://istio.io/latest/docs/tasks/security/authorization/authz-custom/ , should I be looking elsewhere?

Comment (bochengchu) [reply features/external_authorization.md:165]: Performance tests are here: https://github.com/istio/tools/tree/master/perf/benchmark/configs/istio/ext-authz

**Comment by linsun** [comment]
@bochengchu thanks for submitting this for review! I think it is reasonable overall.

I do have 1 comment while reading docs that I would like to be addressed before going to beta. The current doc says:

```
Restart Istiod to allow the change to take effect with the following command:

$ kubectl rollout restart deployment/istiod -n istio-system
deployment.apps/istiod restarted

```

Is this restart required?  Seems odd to require restart of istiod after modifying the istio config map.

**Comment by aryan16** [comment]
> @bochengchu thanks for submitting this for review! I think it is reasonable overall.
> 
> I do have 1 comment while reading docs that I would like to be addressed before going to beta. The current doc says:
> 
> ```
> Restart Istiod to allow the change to take effect with the following command:
> 
> $ kubectl rollout restart deployment/istiod -n istio-system
> deployment.apps/istiod restarted
> ```
> 
> Is this restart required? Seems odd to require restart of istiod after modifying the istio config map.

I agree we don’t require a restart, but we are updating the config map and if something is broken in the config, istiod restart would catch that.

**Comment by linsun** [comment]
> > @bochengchu thanks for submitting this for review! I think it is reasonable overall.
> > I do have 1 comment while reading docs that I would like to be addressed before going to beta. The current doc says:
> > ```
> > Restart Istiod to allow the change to take effect with the following command:
> > 
> > $ kubectl rollout restart deployment/istiod -n istio-system
> > deployment.apps/istiod restarted
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Is this restart required? Seems odd to require restart of istiod after modifying the istio config map.
> 
> I agree we don’t require a restart, but we are updating the config map and if something is broken in the config, istiod restart would catch that.

I don't think we require restart Istiod anywhere in our doc these days. Requiring restart is a yellow flag to promote this to beta. I think we should test with no restart and ensure it can work.

**Comment by bochengchu** [comment]
> > > @bochengchu thanks for submitting this for review! I think it is reasonable overall.
> > > I do have 1 comment while reading docs that I would like to be addressed before going to beta. The current doc says:
> > > ```
> > > Restart Istiod to allow the change to take effect with the following command:
> > > 
> > > $ kubectl rollout restart deployment/istiod -n istio-system
> > > deployment.apps/istiod restarted
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Is this restart required? Seems odd to require restart of istiod after modifying the istio config map.
> > 
> > 
> > I agree we don’t require a restart, but we are updating the config map and if something is broken in the config, istiod restart would catch that.
> 
> I don't think we require restart Istiod anywhere in our doc these days. Requiring restart is a yellow flag to promote this to beta. I think we should test with no restart and ensure it can work.

I have tested it and it works as expected without restart.

File: features/external_authorization.md
Comment (ericvn) [reply features/external_authorization.md:165]: @therealmitchconnors Any other feedback?

**Comment by bochengchu** [comment]
PTAL @linsun @ericvn @therealmitchconnors

**Comment by ericvn** [comment]
LGTM.

File: features/external_authorization.md
Comment (therealmitchconnors) [reply features/external_authorization.md:165]: I spoke with @bochengchu offline, and it looks like we need to push a change to docs to include performance expectations.

**Comment by linsun** [comment]
LGTM, thanks for addressing my comments earlier @bochengchu !

File: features/external_authorization.md
Comment (ericvn) [reply features/external_authorization.md:165]: @therealmitchconnors Does https://github.com/istio/istio.io/pull/12176 satisfy this?

Comment (therealmitchconnors) [reply features/external_authorization.md:165]: I think so, though we may want to consider whether we should have istio.io docs pointing to GitHub.  But one way or another that content should be available, which seems to satisfy this.

**Review by therealmitchconnors — APPROVED** [review]

**Review by ericvn — APPROVED** [review]
This PR now has 4 approval/LGTM from TOC members so can be merged.

**Pull request merged**
